Hermes/appscale/hermes/stats/subscribers/profile.py

```diff
@@ -47,30 +55,380 @@ def receive(self, nodes_stats_dict):
+    order of columns.
+
+    Args:
+      include_lists: an instance of IncludeLists describing which fields
+        of processes stats should be written to CSV log.
+    """
+    self._header = (
+      ['utc_timestamp']
+       + converter.get_stats_header(proxy_stats.ProxyStats, include_lists)
+    )
+    self._include_lists = include_lists
+    self._directory = path.join(PROFILE_LOG_DIR, 'proxies')
+    self._summary_file_name_template = 'summary-{property}.csv'
+    try:
+      makedirs(self._directory)
+    except OSError as os_error:
+      if os_error.errno == errno.EEXIST and path.isdir(self._directory):
+        pass
+      else:
+        raise
+    self._summary_columns = self._get_summary_columns()
 
   def receive(self, proxies_stats_dict):
-    # TODO
-    pass
+    """ Implements receive method of base class. Saves newly produced
+    cluster proxies stats to a list of CSV files.
+    One detailed file for each proxy on every node (normally one LB node)
+    and 3 summary files.
+
+    Args:
+      proxies_stats_dict: a dict with node IP as key and list of
+        ProxyStatsSnapshot as value.
+    """
+    services_summary = collections.defaultdict(self.ServiceProxySummary)
+
+    for node_ip, snapshots in proxies_stats_dict.iteritems():
+
+      if snapshots:
+        # Add summary using only the latest snapshot from the node
+        for proxy in snapshots[-1].proxies_stats:
+          # Add this proxy stats to service summary
+          service_name = proxy.unified_service_name
+          if proxy.application_id:
+            service_name = '{}-{}'.format(service_name, proxy.application_id)
+          summary = services_summary[service_name]
+          summary.requests_rate += proxy.frontend.req_rate
+          summary.bytes_in_out += proxy.frontend.bin + proxy.frontend.bout
+          summary.errors += proxy.frontend.hrsp_4xx + proxy.frontend.hrsp_5xx
+
+      # Write detailed proxy stats using all snapshots received from the node
+      for snapshot in snapshots:
+        for proxy in snapshot.proxies_stats:
+          # Write stats of the specific proxy to its CSV file
+          with self._prepare_file(node_ip, proxy.name) as csv_file:
+            writer = csv.writer(csv_file)
+            row = (
+              [snapshot.utc_timestamp]
+              + converter.stats_to_list(proxy, self._include_lists)
+            )
+            writer.writerow(row)
+
+    # Update self._summary_columns ordered dict (set)
+    for service_name in services_summary:
+      if service_name not in self._summary_columns:
+        self._summary_columns.append(service_name)
+
+    # Write summary
+    self._save_summary(services_summary)
+
+  def _prepare_file(self, node_ip, pxname):
+    """ Prepares CSV file with name <node-IP>/<pxname>.csv
+    for appending new lines.
+
+    Args:
+      node_ip: a string representation of load balancer node IP
+      pxname: a string name of proxy as it's shown haproxy stats
```

Comment:
We can. pxname - is name which used in haproxy stats. So I decided to use it as it is.